core/trino-main/src/main/java/io/trino/operator/function/StreamingTableFunctionProcessor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.function;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Ints;
+import io.trino.operator.WorkProcessor;
+import io.trino.spi.Page;
+
+import java.util.List;
+import java.util.Optional;
+
+public class StreamingTableFunctionProcessor
+        implements WorkProcessor.Transformation<Page, List<Optional<Page>>>
+{
+    private boolean processEmptyInput;
+    private final int[][] requiredChannels;
+
+    public StreamingTableFunctionProcessor(boolean processEmptyInput, List<List<Integer>> requiredChannels)
+    {
+        this.processEmptyInput = processEmptyInput;
+        this.requiredChannels = requiredChannels.stream()
+                .map(Ints::toArray)
+                .toArray(int[][]::new);
+    }
+
+    @Override
+    public WorkProcessor.TransformationState<List<Optional<Page>>> process(Page input)
+    {
+        if (input == null) {
+            if (processEmptyInput) {
+                // it can only happen at the first call to process(), which implies that there is no input. Empty PagesIndex can be passed on only once.
+                processEmptyInput = false;
+                return WorkProcessor.TransformationState.ofResult(null, false);
+            }
+            return WorkProcessor.TransformationState.finished();
+        }
+
+        // there is input, so we are not interested in processing empty input
+        processEmptyInput = false;
+
+        ImmutableList.Builder<Optional<Page>> sourcePages = ImmutableList.builder();
+
+        for (int[] requiredChannel : requiredChannels) {
```

Comment:
`requiredChannel` is multiple channels. Also, what is it required for? Could you rename it so it is more meaningful?

Reply:
`requiredChannels` maps the required set of blocks for each table function argument which is of TableArgumentSpecification type. cc: @kasiafi Please correct if my understanding is wrong.

Reply:
Correct, `requiredChannels` describes which inputs the function requires from each of its table arguments. The naming is consistent with table function analysis ([here](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/function/table/TableFunctionAnalysis.java#L51)).